--- a/libs/context/build/Jamfile.v2
+++ b/libs/context/build/Jamfile.v2
@@ -116,7 +116,7 @@

 actions masm
 {
-    ml /c /Fo"$(<)" "$(>)"
+    ml /safeseh /c /Fo"$(<)" "$(>)"
 }

 actions masm64
--- a/tools/build/src/tools/msvc.jam
+++ b/tools/build/src/tools/msvc.jam
@@ -1117,7 +1117,7 @@
         # Turn on some options for i386 assembler
         #  -coff  generate COFF format object file (compatible with cl.exe output)
         local default-assembler-amd64 = ml64 ;
-        local default-assembler-i386  = "ml -coff" ;
+        local default-assembler-i386  = "ml -coff -safeseh" ;
         local default-assembler-ia64  = ias ;
         local default-assembler-ia64  = armasm ;
